sdk/eventhub/Microsoft.Azure.EventHubs/src/Amqp/AmqpEventDataSender.cs

```diff
@@ -81,6 +81,15 @@ protected override async Task OnSendAsync(IEnumerable<EventData> eventDatas, str
                         {
                             throw AmqpExceptionHelper.ToMessagingContract(amqpException.Error);
                         }
+                        catch (Exception ex)
+                        {
+                            if (AmqpExceptionHelper.TryConvertToRetriableException(ex, out var retriableEx))
+                            {
+                                throw retriableEx;
```

Comment:
Should we consider using `ExceptionDispatchInfo.Capture(retriableEx).Throw()`  to preserve details?

Reply:
This doesn't look necessary since it is not rethrowing the original exception. Code is throwing brand new EventHubsException with original exception as the innerException. There is no chance of losing stacktrace on inner exception.